src/pyNAVIS/plots.py

```diff
@@ -385,3 +386,323 @@ def difference_between_LR(spikes_file, settings, return_data = False, graph_tile
+        ax.set_xlim([0, localization_settings.mso_num_neurons_channel])
+        ax.set_ylim([0, localization_file.mso_timestamps[-1]])
+        ax.set_zlim([0, settings.num_channels])
+
+        plt.tight_layout()
+        msospk_fig.show()
+
+    @staticmethod
+    def mso_localization(localization_file, settings, localization_settings, graph_title = "MSO localization estimation", start_at_zero = True, verbose = False):
+        """
+        Plots the result of the coincidence counters of the Jeffress model for the MSO according to the activity in the LocalizationFile.
+
+        This is, the neuron that fired the most in a time bin, thus indicating the sound source position.
+
+        Parameters:
+                localization_file (LocalizationFile): Localization file to plot.
+                settings (MainSettings): Configuration parameters for the file to plot.
+                localization_settings (LocalizationSettings): Localization configuration parameters for the file to plot.
+                graph_title (string, optional): Text that will appear as title for the graph.
+                start_at_zero (boolean, optional): If set to True, the X axis will start at 0, instead of starting at the minimum timestamp.
+                verbose (boolean, optional): Set to True if you want the execution time of the function to be printed.
+        
+        Returns:
+                None.
+
+        Note:
+                None.
+        """
+
+        if verbose == True: start_time = time.time()
+
+        last_time = 0
+
+        # Set the timestamps limits according to start_at_zero option
+        if start_at_zero:
+            total_time = max(localization_file.mso_timestamps)
+            last_time = 0
+        else:
+            total_time = max(localization_file.mso_timestamps) - min(localization_file.mso_timestamps)
+            last_time = min(localization_file.mso_timestamps)
+
+        # Estimate the number of time bins
+        its = int(math.ceil(total_time/settings.bin_size))
+
+        # Create the activity array
+        mso_activity = np.zeros((its, localization_settings.mso_num_neurons_channel))
+        mso_max_activity = np.zeros(its)
+
+        # For each time bin, calculate the activity of all the MSO neurons and get the winner
+        for i in range (0, its):
+            a = bisect_left(localization_file.mso_timestamps, last_time)
+            b = bisect_right(localization_file.mso_timestamps, last_time + settings.bin_size)
+
+            blockNeuronIDs = localization_file.mso_neuron_ids[a:b]
+
+            spikes = np.bincount(blockNeuronIDs, minlength=localization_settings.mso_num_neurons_channel)
+
+            last_time += settings.bin_size
+
+            mso_activity[i, :] = spikes
+
+            index_max_activity = np.argmax(mso_activity[i])
+
+            mso_max_activity[i] = index_max_activity
+
+        if verbose == True: print('ACTIVITY MATRIX CALCULATION', time.time() - start_time)
```

Comment:
Change to MSO_LOCALIZATION PLOT CALCULATION

Reply:
Done!